lib/services/access_checker.go

```diff
@@ -298,6 +307,32 @@ func (a *accessChecker) CheckAccess(r AccessCheckable, mfa AccessMFAParams, matc
 	return trace.Wrap(a.RoleSet.checkAccess(r, mfa, matchers...))
 }
 
+// GetKubeResources returns the allowed and denied Kubernetes Resources configured
+// for a user.
+func (a *accessChecker) GetKubeResources(cluster types.KubeCluster) ([]types.KubernetesResource, []types.KubernetesResource) {
+	var (
+		allowedResources []types.KubernetesResource
+		deniedResources  []types.KubernetesResource
+	)
+	if len(a.info.AllowedResourceIDs) > 0 {
+		for _, r := range a.info.AllowedResourceIDs {
+			if r.Kind == types.KindKubePod && r.Name == cluster.GetName() && r.ClusterName == a.localCluster {
+				splitted := strings.SplitN(r.SubResourceName, "/", 2)
+				allowedResources = append(allowedResources,
+					types.KubernetesResource{
+						Kind:      r.Kind,
+						Namespace: splitted[0],
+						Name:      splitted[1],
+					},
+				)
+			}
+		}
+		return allowedResources, deniedResources
```

Comment:
Is deniedResources always supposed to be empty here, or did we miss appending to it above?

If it is supposed to be empty, let's make that clearer.

Reply:
Yes, it's supposed to be empty when the user is using a resource access request.
I added a comment to make it clear